letsencrypt/cli.py

```diff
@@ -740,6 +738,52 @@ def install(config, plugins):
     le_client.enhance_config(domains, config)
 
 
+def _set_by_cli(var):
+    """
+    Return True if a particular config variable has been set by the user
+    (CLI or config file) including if the user explicitly set it to the
+    default.  Returns False if the variable was assigned a default value.
+    """
+    detector = _set_by_cli.detector
+    if detector is None:
+        # Setup on first run: `detector` is a weird version of config in which
+        # the default value of every attribute is wrangled to be boolean-false
+        plugins = plugins_disco.PluginsRegistry.find_all()
+        # reconstructed_args == sys.argv[1:], or whatever was passed to main()
+        reconstructed_args = _parser.args + [_parser.verb]
+        default_args = prepare_and_parse_args(plugins, reconstructed_args, detect_defaults=True)
+        detector = _set_by_cli.detector = configuration.NamespaceConfig(default_args, fake=True)
```

Comment:
What is the benefit of creating a `NamespaceConfig` and the accompanying changes versus just using the return value of `prepare_and_parse args`? Seems like unnecessary wrapping/complexity.